### PR TITLE
REP-16 Replications UI, Site enhancements

### LINKED ISSUE #16: As an administrator, I'd like to be able to configure Replications through the UI
**Is your feature request related to a problem? Please describe.**
Currently configuration is done through the command line. It is prone to error and has low usability.

**Describe the solution you'd like**
A form through the UI to create Replications.

**Describe alternatives you've considered**
N/A

**Additional context**
Mockups: https://balsamiq.cloud/swxif/pfb6i6i/rB287

### PR DESCRIPTION
**Pull request by peterhuffer**
#### What does this PR do?
- Adds form for adding Replications
- Table for viewing and deleting Replications
- Delete button for sites
- Styling improvements and refactors.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@mcalcote @clockard @kcover 

#### How should this be tested? (List steps with links to updated documentation)
- Create a Replication, verify UI auto updates to table view.
- Delete the replication, verify UI auto updates back to main home view.
- Confirm creating and deleting Sites.

#### What are the relevant tickets?
Fixes #16 

#### Screenshots (if appropriate)
repmainhome
repcreateform

Normal table view
image

Table view with overflow
overflow
sitesdelete

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: ui/src/main/webapp/components/sites/site.spec.js [hunk 1]
```diff
@@ -1,26 +1,10 @@
-/*global test, mount, expect */
+/*global test, shallow, expect */
 import React from 'react'
 import Site from './Site'
 
 test('site renders correctly', () => {
-  const wrapper = mount(
-    <Site name='example' connected={true} onClick={() => {}} />
+  const wrapper = shallow(
+    <Site name='example' content='www.example.com' id='1234' />
```
Comment (kcover) [inline ui/src/main/webapp/components/sites/site.spec.js]: ❓ Do we have any other automated testing right now?

File: ui/src/main/webapp/components/replications/AddReplication.js [hunk 2]
```diff
@@ -286,6 +296,8 @@ class AddReplication extends React.Component {
                   error.graphQLErrors.forEach(e => {
                     if (e.message === 'INVALID_FILTER') {
                       this.handleInvalidFilter()
+                    } else if (e.message === 'DUPLICATE_CONFIGURATION') {
+                      this.handleInvalidName()
```
Comment (kcover) [inline ui/src/main/webapp/components/replications/AddReplication.js]: ❓ I think Chris' PR had errors for having the same source and destination, and for trying to use a site that hasn't been created yet. Does the UI have a way of handling those situations?

Comment (peterhuffer) [reply ui/src/main/webapp/components/replications/AddReplication.js]: The UI will restrict the sites that can be chosen to be only what is configured. It also makes sure the same site cannot be selected for both source and destination.

File: ui/src/main/webapp/components/sites/site.spec.js (hunk 1 above)
Comment (peterhuffer) [reply ui/src/main/webapp/components/sites/site.spec.js]: Nope :D

File: ui/package.json [hunk 3]
```diff
@@ -72,10 +76,10 @@
     ],
     "coverageThreshold": {
       "global": {
-        "statements": 5,
-        "branches": 11,
-        "functions": 7,
-        "lines": 5
```
[the remark this answers is not in the document]
Comment (peterhuffer) [reply ui/package.json]: Global coverage numbers like jacoco.

**Comment by kcover** [comment]
Successful Hero! 🎉 tested everything including table overflow and deleting sites that are in use.

**Pull request merged**